api/client/src/test/java/org/projectnessie/client/auth/oauth2/TestOAuth2Client.java

```diff
@@ -115,6 +118,16 @@ void testBackgroundRefresh() throws Exception {
         token = client.authenticate();
         assertThat(token.getPayload()).isEqualTo("access-refreshed");
         assertThat(client.getCurrentTokens()).isInstanceOf(RefreshTokensResponse.class);
+
+        // emulate executor running the scheduled renewal task and detecting that the client is idle
+        // after 30+ seconds of inactivity
+        i.set(i.get().plusSeconds(31));
+        currentRenewalTask.get().run();
+        assertThat(client.sleeping).isTrue();
+
+        // should exit sleeping mode on next authenticate() call
+        client.authenticate();
```

Comment:
Do you think we could add a test case for task scheduling failures? Those may happen with user-supplied executors (e.g. short queue), I guess... WDYT?

Reply:
Yes, good idea. Do you mind if I tackle the scheduling failures stuff in a follow-up PR?